framework/PostProcessors/Metric.py

```diff
@@ -0,0 +1,289 @@
+        inputType = currentInput.type
+
+      if isinstance(currentInput, Files.File):
+        self.raiseAWarning(self, "Input type '", inputType, "' is not yet implemented. This input will be skipped")
+      elif inputType == 'HDF5':
+        self.raiseAWarning(self, "Input type '", inputType, "' is not yet implemented. This input will be skipped")
+      elif inputType == 'PointSet':
+        for feature in self.features:
+          if feature in currentInput.getParaKeys('input'):
+            if feature in inputDict['features'].keys():
+              self.raiseAnError(IOError, "Same feature variable '", feature, "' is found in multiple input objects")
+            inputDict['features'][feature] = currentInput.getParam('input', feature, nodeId = 'ending')
+          elif feature in currentInput.getParaKeys('output'):
+            if feature in inputDict['features'].keys():
+              self.raiseAnError(IOError, "Same feature variable '", feature, "' is found in multiple input objects")
+            inputDict['features'][feature] = currentInput.getParam('output', feature, nodeId = 'ending')
+        if self.targets:
+          for target in self.targets:
+            if target in currentInput.getParaKeys('input'):
+              if target in inputDict['targets'].keys():
+                self.raiseAnError(IOError, "Same target variable '", target, "' is found in multiple input objects")
+              inputDict['targets'][target] = currentInput.getParam('input', target, nodeId = 'ending')
+            elif target in currentInput.getParaKeys('output'):
+              if target in inputDict['targets'].keys():
+                self.raiseAnError(IOError, "Same target variable '", target, "' is found in multiple input objects")
+              inputDict['targets'][target] = currentInput.getParam('output', target, nodeId = 'ending')
+      elif inputType == 'HistorySet':
+        self.dynamic = True
+        self.raiseAnError(IOError, "Metric can not process HistorySet, because this capability is not implemented yet")
+
+    unidentifiedVars = set(self.features) - set(inputDict['features'].keys())
+    if len(unidentifiedVars) > 0:
+      self.raiseAnError(IOError, "Features: '", str(unidentifiedVars), "' could not be found in the input objects")
+    unidentifiedVars = set(self.targets) - set(inputDict['targets'].keys())
+    if len(unidentifiedVars) > 0:
+      self.raiseAnError(IOError, "Targets: '", str(unidentifiedVars), "' could not be found in the input objects")
+
+    return inputDict
+
+  def initialize(self, runInfo, inputs, initDict) :
+    """
+      Method to initialize the pp.
+      @ In, runInfo, dict, dictionary of run info (e.g. working dir, etc)
+      @ In, inputs, list, list of inputs
+      @ In, initDict, dict, dictionary with initialization options
+    """
+    PostProcessor.initialize(self, runInfo, inputs, initDict)
+    self.__workingDir = runInfo['WorkingDir']
+
+  def _localReadMoreXML(self, xmlNode):
```

Comment:
Should use InputData to do this.  (If you want, I can do the commit for this)

Reply:
Thanks, let me learn how to add it first. This is a good chance for me to understand the structure of InputData.

Reply:
Switched to InputData.

Reply:
Okay, thanks.